numba/cuda/cudadrv/devicearray.py

```diff
@@ -146,11 +146,26 @@ def copy_to_device(self, ary, stream=0):
         if ary.size == 0:
             # Nothing to do
             return
+        sentry_contiguous(self)
         stream = self._default_stream(stream)
         if _driver.is_device_memory(ary):
+            if self.ndim > 1 and self.flags['C_CONTIGUOUS'] != ary.flags['C_CONTIGUOUS']:
```

Comment:
You don't need the `self.ndim > 1`.  Strided 1D array (e.g. `np.arange(20)[::2]`) is not contiguous.

Reply:
for `self.flags['C_CONTIGUOUS'] != ary.flags['C_CONTIGUOUS']`, I am wondering if it is a stronger check with `self.strides != ary.strides`.